### PR TITLE
Added area of Trapezoid.

### PR DESCRIPTION
**Pull request by devesh-0419**
A quadrilateral with at least one pair of parallel sides is called a trapezoid.
Area of trapezoid = ((sum of length of two parallel side)/2)*height
The area can be a double so the input type is double and it consist of two bases so I named it base1 and base2 for convenience...
[trapezoid](https://en.wikipedia.org/wiki/Trapezoid)

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

**Review by Panquesito7 — DISMISSED** [review]
Looks good. Thanks! 🎉

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Please fix the [`clang-tidy`](https://github.com/TheAlgorithms/C-Plus-Plus/actions/runs/3200854359/jobs/5228233552) warnings. Let us know if you need any help. 🙂

**Comment by devesh-0419** [comment]
Thanks, I am fixing the code

**Comment by devesh-0419** [comment]
please review my code. Fixed the bug that was mentioned earlier

File: math/area.cpp [hunk 1]
```diff
@@ -76,6 +76,20 @@ T parallelogram_area(T base, T height) {
     return base * height;
 }
 
+ /**
+ * @brief area of a
+ * [trapezoid/trapezium with two parallel
+ * side](https://en.wikipedia.org/wiki/Trapezoid) ((b1 + b2)/2]*h)
+ * @param base1 is the length of one of the parallel side of the trapezium
```
Comment (Panquesito7) [inline math/area.cpp:83]: You may want to describe the `T` param as well.
```suggestion
 * @tparam <description>
 * @param base1 is the length of one of the parallel side of the trapezium
```

File: math/area.cpp [hunk 2]
```diff
@@ -76,6 +76,20 @@ T parallelogram_area(T base, T height) {
...
+ * @param base2 is the length of other parallel side of the trapezium
+ * @param height is the length between both parallel side
```
Comment (krishnacx) [inline math/area.cpp:85]: can be reworded to "between parallel sides"

**Pull request closed without merging**